pkg/datapath/loader/loader.go

```diff
@@ -321,6 +321,20 @@ func (l *Loader) reloadDatapath(ctx context.Context, ep datapath.Endpoint, dirs
 			}
 			return err
 		}
+		if ep.RequireEgressProg() {
```

Comment:
Have you considered attaching bpf_host.o instead to a master device? This would reduce a security concern by a bit as privileged pods won't be able to detach the program.

Reply:
You are right, but some scenarios may not have master NIC, such as eni of vpc, sriov of physical network, etc.

Reply:
Not sure about the SRIOV case, but for the ENI you can use the generic chainer - https://docs.cilium.io/en/v1.9/gettingstarted/cni-chaining-aws-cni/.